planner/core/plan_cache.go

```diff
@@ -442,7 +443,7 @@ func rebuildRange(p Plan) error {
 				if err != nil {
 					return err
 				}
-				if !isSafeRange(x.AccessConditions, ranges, false, nil) {
+				if len(ranges.Ranges) != 1 || !isSafeRange(x.AccessConditions, ranges, false, nil) {
```

Comment:
Add more restrictions for safety.

Reply:
For example, when the first execution, we got `[[1], [2]]` from `where k in (1, 2)`, and when the second execution, we got `[[1]]` from `where k in (1, 1)`. For safety, if the range lengths changed, we generate a new plan instead of re-using the old one.